src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -107,28 +94,22 @@ public async Task<ImmutableArray<DiagnosticData>> GetProjectDiagnosticsAsync(Can
                     project, documentIds: [], includeProjectNonLocalResult: true, cancellationToken).ConfigureAwait(false);
             }
 
-            protected override bool ShouldIncludeDiagnostic(DiagnosticData diagnostic)
-                => _diagnosticIds == null || _diagnosticIds.Contains(diagnostic.Id);
-
-            protected override async Task ProduceDiagnosticsAsync(
+            private async Task ProduceDiagnosticsAsync(
                 Project project,
                 IReadOnlyList<DocumentId> documentIds,
                 bool includeProjectNonLocalResult,
                 ArrayBuilder<DiagnosticData> builder,
                 CancellationToken cancellationToken)
             {
-                // get analyzers that are not suppressed.
                 var stateSetsForProject = await StateManager.GetOrCreateStateSetsAsync(project, cancellationToken).ConfigureAwait(false);
                 var stateSets = stateSetsForProject.Where(s => ShouldIncludeStateSet(project, s)).ToImmutableArrayOrEmpty();
 
-                // unlike the suppressed (disabled) analyzer, we will include hidden diagnostic only analyzers here.
-                var compilation = await CreateCompilationWithAnalyzersAsync(project, stateSets, Owner.AnalyzerService.CrashOnAnalyzerException, cancellationToken).ConfigureAwait(false);
-
-                var result = await Owner.GetProjectAnalysisDataAsync(compilation, project, stateSets, cancellationToken).ConfigureAwait(false);
+                var result = await GetOrComputeDiagnosticAnalysisResultsAsync(stateSets).ConfigureAwait(false);
 
                 foreach (var stateSet in stateSets)
                 {
-                    var analysisResult = result.GetResult(stateSet.Analyzer);
+                    if (!result.TryGetValue(stateSet.Analyzer, out var analysisResult))
+                        continue;
```

Comment:
no need for this pointless wrapper type that only wraps an ImmDict.  It's now just an immdic.